### ISSUE TITLE
Help reproduce the performance

### ISSUE THREAD

**Issue by neuron2018** [xuepengcheng1231/CortiLife-A-Unified-Framework-for-Cortical-Representation-Learning-across-the-Lifespan#1]
Hi authors,

Thank you for sharing this interesting work. I have been reading your paper and trying to reproduce the reported results, but so far I have not been able to match them.

I would really appreciate your help with a few questions:

1. Are the pretrained model weights/checkpoints available, or do you plan to release them?
2. Will the processed dataset splits / preprocessing pipeline be released?
3. Could you share more detailed instructions for reproduction, such as:

   * exact data preprocessing steps
   * train/validation/test split details
   * hyperparameters used for pretraining and fine-tuning
   * number of epochs, batch size, learning rate schedule, and random seeds
   * software environment / package versions
4. Is there any missing configuration, script, or setting in the current repository that is necessary to reproduce the results?
5. Is it a unified foundation framework as mentioned in the paper across the datasets?

For context, I have already:

* set up the environment
* followed the repository instructions
* attempted to run the training / evaluation pipeline
* but the reproduced performance is still noticeably different from the paper

If helpful, I can also share my exact commands, environment, and the results I obtained.

Thanks again for your work and for any guidance you can provide.

**Comment by neuron2018** [comment]
Any response for this reproduce question? We have devoted much effort to it but still cannot figure out how to reproduce it.

**Comment by neuron2018** [comment]
@xuepengcheng1231  Can you please help me with it?

**Comment by xuepengcheng1231** [comment]
Sorry, I just saw these comments. I am currently organizing these environment settings. I will reply to them one by one later. In addition, we will upload a new version for reference.

**Comment by xuepengcheng1231** [comment]
1. Regarding the pretrained weights, we would like to clarify that our current evaluation setting is not a foundation-model-style evaluation in the strict sense. Instead, it follows a “pretraining + finetuning” framework. Therefore, for each new test dataset, pretraining is generally required again. In other words, if you would like to apply this framework to your own dataset, we recommend performing pretraining again using your own data.
2. All cortical data in our study were processed using FreeSurfer recon-all (https://www.sciencedirect.com/science/article/pii/S1053811912000389). The cortical features, including CT, MC, and SD, were extracted through this pipeline and subsequently registered to the fsaverage cortical space for downstream model training and analysis.
3. Regarding the details of the train / validation / test split, the corresponding implementation is already provided in our code. Please refer to the FinetuningSurfClip.py file, where the specific dataset splitting logic and related settings are included.
4. The hyperparameters used for both pretraining and fine-tuning have all been provided in the code. In addition, we have included the training commands used in our experiments. For hyperparameters that are not explicitly specified in the command line, the default values defined in the code are used.
5. Regarding the package version information, we have exported the package versions from our experimental environment and provided them for your reference.

[packages.txt](https://github.com/user-attachments/files/27513223/packages.txt)
6. We sincerely apologize for the missing configuration files. We recently found that some files were not uploaded to the repository. They have now been added, and the code repository has been fully updated accordingly.
7. In fact, our model should not be considered a foundation framework in the strict sense, and we have not claimed that it reaches the level of a foundation model. Our evaluation is only conducted under a pretraining + fine-tuning setting.

Specifically, our method aims to improve generalization through a carefully designed encoder and training on a relatively large-scale dataset. The experimental results suggest that this strategy can achieve good generalization performance across downstream tasks.

However, we acknowledge that, with respect to the concept of a foundation model, the current scale of data used in our work may still be insufficient. Therefore, we would describe our framework more appropriately as a pretraining-based representation learning framework rather than a true foundation model.

**Comment by neuron2018** [comment]
Thanks for the detailed reply.

1. This clarification appears inconsistent with the framing of the ICLR paper. The paper presents CortiLife as a unified framework for cortical representation learning across the lifespan and evaluates it across multiple downstream datasets and tasks. This framing suggests that the pretrained encoder is intended to learn transferable cortical representations, similar to a foundation-model-style pretraining paradigm. If pretraining must be repeated for each new test dataset, then the method is closer to dataset-specific self-supervised pretraining followed by finetuning, rather than a reusable pretrained model.

 This distinction is important because “pretraining + finetuning” and “foundation-model-style transfer” have different evaluation standards. In the former, the target dataset can influence representation learning during pretraining, while in the latter, the pretrained model should be fixed and transferred to unseen datasets or tasks. If CortiLife requires new pretraining for every target dataset, then its reported generalization claims should be interpreted more narrowly, and comparisons to foundation models or reusable pretrained cortical encoders may be overstated.

**Comment by neuron2018** [comment]
2. Thanks for sharing the data. I am looking into it and will circle back then.

**Comment by neuron2018** [comment]
3. Thanks for confirming.

**Comment by neuron2018** [comment]
4. But the hyper-parameters in the code are inconsistent with the ones described in the paper. I will first explore it myself.

**Comment by neuron2018** [comment]
6. Thanks. I'm reading them.

**Comment by neuron2018** [comment]
7. Same as the 1st.

**Comment by xuepengcheng1231** [comment]
Thank you very much for your constructive comments. We indeed identified this issue during our work, and we are currently improving the method while further expanding the dataset.

Regarding the inconsistency in hyperparameters that you mentioned, please refer to the actual implementation in the code as the final reference. Since our code was implemented based on the repository released with the DetailCLIP paper, some hyperparameters may have been modified during the actual implementation and training process, but the corresponding configurations in args were not updated accordingly.

**Comment by neuron2018** [comment]
Thanks for your response. One quick question, can you prepare a readme to ramp up us how to correctly reproduce it?

**Comment by neuron2018** [comment]
@xuepengcheng1231 I still cannot replica the results, can you provide a readme to guide me how to replicate the results?

**Comment by xuepengcheng1231** [comment]
Sorry, I just saw this. What step are you at now? Which step failed to reproduce the results? Without a clear problem description, I don't know where your problem lies. Could you tell me your data details, including other configurations?

**Comment by xuepengcheng1231** [comment]
If you need, i can give you some of my test data and part of the model weight file. You can try it again.

**Comment by neuron2018** [comment]
Can you at lease provide a readme for me?

Here is the cmd I used:

torchrun --nproc_per_node=4 --nnodes=1 TrainSurfClip.py \
  -j 4 \
  --root_path ./data_surfclip \
  --train_image_filename No_ADHD_single.npy \
  --train_text_filename No_ADHD_text_embeddings_single.npy \
  --output-dir ... \
  --mask-ratio 0.75 \
  --weight_mask 0.7 \
  --model SurfClip_512_deeper_10_single_channel
